src/app/modules/terminal/xterm-attach-addon.ts

```diff
@@ -1,32 +1,34 @@
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
 import { Terminal, IDisposable, ITerminalAddon } from 'xterm';
+import { ShellService } from 'app/services/shell.service';
 
 /**
  * This is basically the same as xterm-addon-attach
  * https://github.com/xtermjs/xterm.js/tree/master/addons/xterm-addon-attach
  * but it always sends messages as binary.
  */
+@UntilDestroy()
```

Comment:
UntilDestroy relies on `ngDestroy`, which probably wouldn't work for this class.
You may need to do something else here.